rpc/grpc/server/services/blockresultservice/service.go

```diff
@@ -0,0 +1,89 @@
+package blockresultservice
+
+import (
+	"context"
+
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	"github.com/cometbft/cometbft/libs/log"
+	sm "github.com/cometbft/cometbft/state"
+	"github.com/cometbft/cometbft/store"
+
+	brs "github.com/cometbft/cometbft/proto/tendermint/services/block_results/v1"
+)
+
+type blockResultsService struct {
+	stateStore sm.Store
+	blockStore *store.BlockStore
+	logger     log.Logger
+}
+
+// New creates a new CometBFT block results service server.
+func New(bs *store.BlockStore, ss sm.Store, logger log.Logger) brs.BlockResultsServiceServer {
+	return &blockResultsService{
+		stateStore: ss,
+		blockStore: bs,
+		logger:     logger.With("service", "BlockResultsService"),
+	}
+}
+
+// GetBlockResults returns the block results of the requested height.
+func (s *blockResultsService) GetBlockResults(_ context.Context, req *brs.GetBlockResultsRequest) (*brs.GetBlockResultsResponse, error) {
+	logger := s.logger.With("endpoint", "GetBlockResults")
+	ss, err := s.stateStore.Load()
+	if err != nil {
+		logger.Error("Error loading store", "err", err)
+		return nil, status.Error(codes.Internal, "Internal server error")
+	}
+	if req.Height > ss.LastBlockHeight || req.Height < 0 {
+		logger.Error("Error validating GetBlockResults request height")
```

Comment:
```suggestion
```

Could pollute operators' logs. And the full error's returned to the caller anyways.